File: 293. Flip Game.cpp
class Solution {
public:
    // brute force: O(N), 2.29%, ok
    vector<string> generatePossibleNextMoves(string s) {
        vector<string> ret;
        if(s.empty()) return ret;
        for(int i = 1; i < s.size(); i++){
            if(s[i] == s[i - 1] && s[i] == '+'){
                string temp = s;
                temp[i] = temp[i - 1] = '-';
                ret.push_back(temp);
            }
        }
        return ret;
    }
};

File: 293. Flip Game.py
class Solution(object):
    
    # brute force: O(N) time, O(1) space
    def generatePossibleNextMoves(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        ret = []
        for i in xrange(len(s)-1):
            if s[i] == s[i+1] == '+':
                ret.append(s[:i] + '--' +s[i+2:])
        return ret
        